example_14/hardware/rrbot_sensor_for_position_feedback.cpp

```diff
@@ -0,0 +1,284 @@
+
+  address_.sin_family = AF_INET;
+  address_.sin_addr.s_addr = INADDR_ANY;
+  address_.sin_port = htons(socket_port_);
+
+  RCLCPP_INFO(rclcpp::get_logger("RRBotSensorPositionFeedback"), "Binding to socket address.");
+  if (bind(obj_socket_, reinterpret_cast<struct sockaddr *>(&address_), sizeof(address_)) < 0)
+  {
+    RCLCPP_FATAL(rclcpp::get_logger("RRBotSensorPositionFeedback"), "Binding to socket failed.");
+    return hardware_interface::CallbackReturn::ERROR;
+  }
+
+  // Storage and Thread for incoming data
+  rt_incomming_data_ptr_.writeFromNonRT(std::numeric_limits<double>::quiet_NaN());
+  incoming_data_thread_ = std::thread(
+    [this]()
+    {
+      // Await and accept connection
+      RCLCPP_INFO(
+        rclcpp::get_logger("RRBotSensorPositionFeedback"), "Listening for connection on port %d.",
+        socket_port_);
+      if (listen(obj_socket_, 1) < 0)
+      {
+        RCLCPP_FATAL(
+          rclcpp::get_logger("RRBotSensorPositionFeedback"), "Cannot listen from the server.");
+        return hardware_interface::CallbackReturn::ERROR;
+      }
+
+      sock_ = accept(
+        obj_socket_, reinterpret_cast<struct sockaddr *>(&address_),
+        reinterpret_cast<socklen_t *>(&address_length_));
+      if (sock_ < 0)
+      {
+        RCLCPP_FATAL(
+          rclcpp::get_logger("RRBotSensorPositionFeedback"), "Cannot accept on the server.");
+        return hardware_interface::CallbackReturn::ERROR;
+      }
+      RCLCPP_INFO(rclcpp::get_logger("RRBotSensorPositionFeedback"), "Accepting on socket.");
+
+      int incoming_data_read_rate = 1000;  // Hz
+      RCLCPP_INFO(
+        rclcpp::get_logger("RRBotSensorPositionFeedback"),
+        "Creating thread for incoming data and read them with %d Hz to not miss any data.",
+        incoming_data_read_rate);
+
+      // Variables for reading from a socket
+      const size_t reading_size_bytes = 1024;
+      char buffer[reading_size_bytes] = {0};
+
+      // Use nanoseconds to avoid chrono's rounding
+      std::this_thread::sleep_for(std::chrono::nanoseconds(1000000000 / incoming_data_read_rate));
+
+      RCLCPP_INFO(rclcpp::get_logger("RRBotSensorPositionFeedback"), "Receiving data");
+      while (rclcpp::ok())
+      {
+        if (recv(sock_, buffer, reading_size_bytes, 0) > 0)
+        {
+          RCLCPP_DEBUG(
+            rclcpp::get_logger("RRBotSensorPositionFeedback"),
+            "Read form buffer sockets data: '%s'", buffer);
+
+          rt_incomming_data_ptr_.writeFromNonRT(std::stod(buffer));
+        }
+        else
+        {
+          RCLCPP_INFO(
+            rclcpp::get_logger("RRBotSensorPositionFeedback"),
+            "Data not yet received from socket.");
+          rt_incomming_data_ptr_.writeFromNonRT(std::numeric_limits<double>::quiet_NaN());
+        }
+
+        bzero(buffer, reading_size_bytes);
+        std::this_thread::sleep_for(std::chrono::nanoseconds(1000000000 / incoming_data_read_rate));
+      }
+    });
```

Comment:
gcc is throwing a warning telling that this lambda funtion is not a ```void function()```. It is probably due to the previous return statement (line 168) which are forcing the non-return type. It would be nice to correct this at the end of the function.
```return hardware_interface::CallbackReturn::SUCCESS; ```